### PR TITLE
Enforce strict locking of lots (vs forced availability)

### PR DESCRIPTION
**Pull request by clonedagain**
Optional ~~module~~ feature to strictly enforce the blocking of lots defined in `stock_lock_lot`, by refusing to move force-assigned goods.
A checkbox lets users ~~install this module~~ disable this strict checking from the warehouse.

### CODE REVIEW

**Comment by clonedagain** [comment]
@pedrobaeza @lbellier it looks like I forgot to make a PR for this one too.

File: stock_lock_lot/views/res_config_view.xml [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+        <record id="view_stock_configuration" model="ir.ui.view">
+            <field name="name">Stock settings: Strict locking</field>
+            <field name="model">stock.config.settings</field>
+            <field name="inherit_id" ref="stock.view_stock_config_settings" />
+            <field name="arch" type="xml">
+                <data>
+                    <xpath expr="//group[last()]" position="after">
+                        <group>
+                            <label for="id" string="Lot blocking" />
+                            <div>
```
Comment (pedrobaeza) [inline stock_lock_lot/views/res_config_view.xml]: Why 2 nested `<div>`?

**Comment by pedrobaeza** [comment]
Isn't better to embed this functionality in the same module instead of creating a new one? The configuration can remain the same, but writing a parameter instead of installing a module.

**Comment by clonedagain** [comment]
I like the separation of concerns very-small modules give (I focus best this way), but I can always change that if it's blocking.

**Comment by pedrobaeza** [comment]
Well, in this case I see very little added functionality, so it makes sense to have both merged in one module (to not overhead this little functionality with manifest, README, views, models...).

**Comment by clonedagain** [comment]
OK why not, do you have an example of how to read the config from the main code please?

**Comment by elicoidal** [comment]
LGTM (no opinion on the granularity question though)

**Comment by pedrobaeza** [comment]
@clonedagain, here you have an example: https://github.com/OCA/manufacture/pull/108/files#diff-f8b1a7cda1ede9e1055014b4a12dfc2eR16

**Comment by clonedagain** [comment]
@pedrobaeza I've squashed the module into stock_lock_lot as you suggested

**Comment by pedrobaeza** [comment]
:+1:

**Comment by clonedagain** [comment]
Rebased to resolve conflicts.

**Comment by lbellier** [comment]
LGTM :+1:

**Pull request merged**